pinot-spi/src/main/java/org/apache/pinot/spi/filesystem/PinotFSFactory.java

```diff
@@ -37,6 +37,7 @@ private PinotFSFactory() {
   }
 
   public static final String LOCAL_PINOT_FS_SCHEME = "file";
+  public static final String RECURSIVE_BLOB_PREFIX = "glob:**";
```

Comment:
recursive hint from glob pattern should do recursive search whenever there's a `**` pattern included, b/c `**` can occur in the middle such as `subFolder/foo/**/bar`

Reply:
Updated the code and added a test.